clang/test/CodeGenSYCL/intel-fpga-nofusion.cpp

```diff
@@ -0,0 +1,61 @@
+// RUN: %clang_cc1 -triple spir64-unknown-unknown-sycldevice -disable-llvm-passes -fsycl -fsycl-is-device -emit-llvm %s -o - | FileCheck %s
+
+// CHECK: br label %while.cond, !llvm.loop ![[MD_NF_1:[0-9]+]]
+// CHECK: br label %for.cond3, !llvm.loop ![[MD_NF_2:[0-9]+]]
+// CHECK: i1 %cmp18, label %do.body, label %do.end, !llvm.loop ![[MD_NF_3:[0-9]+]]
+// CHECK: br label %for.cond20, !llvm.loop ![[MD_NF_4:[0-9]+]]
+// CHECK: br label %for.cond41, !llvm.loop ![[MD_NF_5:[0-9]+]]
+// CHECK: br label %for.cond50, !llvm.loop ![[MD_NF_6:[0-9]+]]
+
+#include "Inputs/sycl.hpp"
+
+void nofusion() {
+  int a[10];
+
+  int i = 0;
+  [[intel::nofusion]] while (i < 10) {
+    a[i] += 7;
+  }
+
+  for (int i = 0; i < 10; ++i) {
+    [[intel::nofusion]] for (int j = 0; j < 10; ++j) {
+      a[i] += a[j];
+    }
+  }
+
+  [[intel::nofusion]] do {
+    a[i] += 4;
+  }
+  while (i < 10)
+    ;
+
+  [[intel::nofusion]] for (int i = 0; i < 10; ++i) {
+    for (int j = 0; j < 10; ++j) {
+      a[i] += a[j];
+    }
+  }
+
+  int k = 0;
+  [[intel::nofusion]] for (auto k : a) {
+    k += 2;
+  }
+
+  [[intel::nofusion]] for (int i = 0; i < 10; ++i) {
+    a[i] += 3;
+  }
+}
+
+int main() {
+  cl::sycl::kernel_single_task<class kernel_function>([]() {
```

Comment:
Can we use `handler::single_task` from the mock header? And also include this mock header as system one?
It doesn't affect tests added in this new patch, but it affects tests for warnings and we want to make sure that all tests are aligned in the same way. It will save time on review and help us to make sure that we test warnings properly because sometimes people copy some old test to make a new one.

Reply:
Tests are updated.